scala/tempo/src/main/scala/com/databrickslabs/tempo/rangeStats.scala

```diff
@@ -0,0 +1,61 @@
+package com.databrickslabs.tempo
+
+import org.apache.spark.sql.expressions.{Window, WindowSpec}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{Column, DataFrame}
+
+/**
+  * The following object has methods for adding rolling statistics based on the timestamp as the base point. The parameter to range stats is a number of seconds to look back to aggregate statistics such as mean, count, sum, min, max, etc.
+  */
+object rangeStats {
```

Comment:
These are great! I'm sure this list will grow over time. That said, since some of these are somewhat costly to calculate, perhaps you could shove the values into delta / df col metadata...this way, if a user is using tempo with the io package you could reference pre-calc'd stats from last run and only re-calc on new data. Often saves a lot of time on large datasets.